File: copy-parser.rb
require 'json'

file_names = Dir['copy/*.json']
lang = {}

file_names.each do |file_name|
  object = JSON.parse(File.read(file_name))
  lang[object.keys[0]] = object
end

puts lang

# turn puts to logger.warn
def copy(key, locals)
  locale = 'en' # this will be dynamic from the session etc
  value = lang[locale][key]

  unless value
    puts "Copy not found, unknown key #{key}"
    return
  end

  if value.is_a? Array
    if value[0].expression
      if !locals
        # defaultObject = _.find(value, { expression: 'true' });
        return compileString(default_object.value)
      end

      # targetObject = _.find(value, function(el) {
      #   return parseExpression(el.expression) === true;
      # });

      if !target_object
        # defaultObject = _.find(value, { expression: 'true' });
        if default_object
          return compileString(default_object.value)
        end

        return ''
      end

      return compileString(target_object.value) # (locals)
    end

    puts "Array without expression requested for copy key: #{key}"
    return ""
  elsif value.is_a? Object
    # write this logic
  end

  def parseExpression(target)
    # class eval here
  end

  def compileString(target)
    # Mustache compile here
  end
end

File: copy-parser.js
var fs = require('fs'),
    util = require('util'),
    lang = {};

fs.readdirSync('./copy').filter((a) => {
    return a.slice(-5) === '.json';
}).forEach((fileName) => {
    var object = JSON.parse(fs.readFileSync('./copy/' + fileName, {encoding: 'utf8'}));
    lang[Object.keys(object)[0]] = object;
});

window.$parse = ngParser; // take this to npm load
// import also Mustache

window.t = function(key, locals) { // export this CommonJS way
    // this function is unstable, still work in progress
    // accept markdown?
    // google this: dot notation on a string:
    var locale = "en", // this will be dynamic from the store for example
        value = lang[locale][key];

    if (_.isUndefined(value)) {
        console.warn('Copy not found, unknown key ' + key);
        return;
    }

    if (_.isArray(value)) {
        if (value[0].expression) {
            var defaultObject, targetObject;

            if (_.isUndefined(locals)) {
                defaultObject = _.find(value, { expression: 'true' });
                return compileString(defaultObject.value);
            }

            targetObject = _.find(value, function(el) {
                return parseExpression(el.expression) === true;
            });

            if (_.isUndefined(targetObject)) {
                defaultObject = _.find(value, { expression: 'true' });
                if (defaultObject) {
                    return compileString(defaultObject.value);
                }

                return '';
            }

            return compileString(targetObject.value); //(locals);
        }

        console.warn('Array without expression requested for copy key: ' + key);
        return '';
    } else if (_.isObject(value)) {
        if (value.expression) {
            if (parseExpression(value.expression)) {
                return compileString(value.value);
            }
            return '';
        }

        console.warn('Object without expression requested for copy key: ' + key);
        return '';
    }

    return compileString(value);

    function parseExpression(target) {
        return $parse(target)(_.merge(Store.getState(), locals));
    }

    function compileString(target) {
        return Mustache.render(target, _.merge(Store.getState(), locals));
    }

    // . notation in the string
};
